rules/DowngradePhp70/Rector/Expr/DowngradeUnnecessarilyParenthesizedExpressionRector.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\Expr;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\ArrayDimFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/uniform_variable_syntax
+ *
+ * @see \Rector\Tests\DowngradePhp70\Rector\Expr\DowngradeUnnecessarilyParenthesizedExpressionRector\DowngradeUnnecessarilyParenthesizedExpressionRectorTest
+ */
+final class DowngradeUnnecessarilyParenthesizedExpressionRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Remove parentheses around expressions where they are not necessary thanks to Uniform variable syntax RFC.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+($f)['foo'];
+($f)->foo;
+($f)->foo();
+($f)::$foo;
+($f)::foo();
+($f)();
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$f['foo'];
+$f->foo;
+$f->foo();
+$f::$foo;
+$f::foo();
+$f();
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        // TODO: Make PHPStan allow Expr namespace.
+        $dummyClassNameUse = str_contains(Expr::class, '\\');
+        assert($dummyClassNameUse);
```

Comment:
Ugh we need to add `Expr` into https://github.com/rectorphp/phpstan-rules/blob/df714324fd2839c9f03d39c437fa6b37ccf127a3/src/Rule/RequireRectorCategoryByGetNodeTypesRule.php#L124. Also discovered https://github.com/symplify/symplify/issues/3872 while working around this.